python/tvm/contrib/hexagon/build.py

```diff
@@ -373,9 +370,10 @@ def _run_server_script(self):
         )
         # Enable port forward for RPC server. We forward 9 ports after the rpc_server_port.
         for i in range(0, 10):
+            port = rpc_server_port + i
+            self.forwarded_ports_.append(port)
             subprocess.check_call(
-                self._adb_device_sub_cmd
-                + ["forward", f"tcp:{rpc_server_port+i}", f"tcp:{rpc_server_port+i}"]
+                self._adb_device_sub_cmd + ["forward", f"tcp:{port}", f"tcp:{port}"]
```

Comment:
If a port forward already exists, does `adb` exit with a non-zero return code?  If the `adb` call exits with a zero return code in that case, then there are two processes trying to start port forwards on overlapping ranges would result in double clean-up occurring.

Reply:
Also, I think we should move the `self.forwarded_ports_.append(port)` to be after the subprocess call.  That way, a cleanup that occurs as the stack unwinds (e.g. [the `hexagon_launcher` fixture](https://github.com/apache/tvm/blob/main/tests/python/contrib/test_hexagon/conftest.py#L184)) wouldn't try to clean up the failed port forward.

Reply:
Thank you for the though on this, and no adb doesn't complain about multiple attempts to forward the same port. I decided to be more rigorous with port checking and only forward ports that are not already forwarded by adb or bound by another process. This seems to have greatly improved the stability. PTAL